src/metabase/search/spec.clj

```diff
@@ -217,6 +222,18 @@
        x))
    expr))
 
+(defn- construct-source-where [id-fields]
+  (cond
+    (keyword? id-fields) [:= (add-table :updated id-fields) (add-table :this id-fields)]
+    (boolean? id-fields) [:= :updated.id :this.id]
+    ;; Vector is probably something like `[:concat :field1 "sep" :field2]`; maybe we should switch to more restrictive
+    ;; notation in `:attrs`?
```

Comment:
I don't see why we need the full power of a honeySQL expression here, why not just use `[:field-1 :field-2 ...]` as the attr, and encapsulate the SQL details. This will avoid things getting weird for Elasticsearch etc

Reply:
Ohh that's right, I wasn't thinking straight. So the problem is that it's handled like a honeysql expression somewhere else, I need to change that.

Reply:
For posterity, we decided to keep this as it kept the implementation simpler for now and there are other SQL coupled details in the spec anyway, so we can resolve them all in one tranch later.